### DISCUSSION TITLE
How do I match any path starting with an anchor (#) tag

### DISCUSSION THREAD

**Discussion by mbharadwaj** [fastapi/fastapi#6709] [Questions] [answered]
### Description / Issue

I am trying to all paths that match a string starting with an anchor tag (#). This does not see to work as the Request parsing step seems to do something special for the # and I have not been able to find a work around. Any tips on workarounds and potential hacks will be greatly appreciated.

### Example

```Python
from fastapi import FastAPI

app = FastAPI()

@app.get("/{full_path:path}")
async def catch_all(request: Request, full_path: str):
    print("full_path: "+full_path)
    return {"full_path": full_path})
```

### Description

* Open the browser and call the endpoint `/error_code=404&error_description=Internal%20error`.
* It returns a JSON with `{"full_path":"error_code=404&error_description=Internal error"}`.
* Open the browser and call the endpoint `/$error_code=404&error_description=Internal%20error`.
* It returns a JSON with `{"full_path":"$error_code=404&error_description=Internal error"}`.
* Open the browser and call the endpoint `/#error_code=404&error_description=Internal%20error`.
* It returns a JSON with `{"full_path":""}`.
* But I expected it to return `{"full_path":"#error_code=404&error_description=Internal error"}`.

### Environment

* OS: MacOS M1
* FastAPI Version: 0.65.3
* Python Version: 3.9.5

**Comment by Mause** [comment, marked as answer]
The `#` is a special character in URLs - it indicates the start of the hash (or "fragment"). Browsers do not pass the hash to the server at all. You may be able to escape it, but I'm not sure if that would be reliable

**Comment by mbharadwaj** [comment]
thank you for your quick response.

duh! i had to handle this on the client. got confused.
